src/Covenant/Internal/ASGNode.hs

```diff
@@ -20,7 +20,12 @@ newtype Id = Id Word64
     ( -- | @since 1.0.0
       Eq,
       -- | @since 1.0.0
-      Ord
+      Ord,
+      -- | @since 1.0.0
+      Bounded,
+      -- | We need this for internal reasons, even though this type class is
+      -- terrible.
+      Enum
```

Comment:
I haven't yet noticed the reason for the `Enum`.

Reply:
Partly because of `EnumMap`, and partly because otherwise, `MonadHashCons` and `HashConsT` wouldn't know how to increment `Id`. Admittedly the first one is _slightly_ self-imposed, but the second would be awkward at best.